### PR TITLE
Add currency conversion proposal

### PR DESCRIPTION
**Pull request by And1sS**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/Price.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.prebid.server.bidder;
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/Price.java:1]: bidder.model

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 2]
```diff
@@ -97,61 +87,45 @@ private ExtImpAdview parseExtImp(Imp imp) {
...
-        modifiedImps.set(0, modifyImp(imps.get(0), masterTagId, bidFloor, bidFloorCur));
+        modifiedImps.set(0, modifyImp(imps.get(0), masterTagId, bidFloor));
         return modifiedImps;
     }
 
-    private static Imp modifyImp(Imp imp,
-                                 String masterTagId,
-                                 BigDecimal bidFloor,
-                                 String bidFloorCur) {
+    private static Imp modifyImp(Imp imp, String masterTagId, Price bidFloor) {
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:124]: Maybe floorPrice?

File: src/main/java/org/prebid/server/bidder/Price.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Value;
+
+import java.math.BigDecimal;
+
+@Value(staticConstructor = "of")
+public class Price {
+
+    String currency;
+
+    BigDecimal price;
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/Price.java:12]: Minor. MB priceValue, just to show difference between model and field?

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 4]
```diff
@@ -97,61 +87,45 @@ private ExtImpAdview parseExtImp(Imp imp) {
...
-
-    private BigDecimal resolveBidFloor(BidRequest request, Imp imp, String bidFloorCurrency) {
-        final BigDecimal bidFloor = imp.getBidfloor();
-        final String bidFloorCur = imp.getBidfloorcur();
-
-        return !bidFloorCurrency.equals(bidFloorCur)
-                ? convertBidFloor(bidFloor, bidFloorCur, imp.getId(), request)
-                : bidFloor;
+    private Price resolveBidFloor(Imp imp, BidRequest bidRequest) {
+        final Price initialPrice = Price.of(imp.getBidfloorcur(), imp.getBidfloor());
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:91]: Minor. In method parameters you used `floorPrice` naming. So, for here `initialFloorPrice`

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 5]
```diff
@@ -97,61 +87,45 @@ private ExtImpAdview parseExtImp(Imp imp) {
...
+                .imp(modifyImps(bidRequest.getImp(), masterTagId, bidFloor))
                 .cur(Collections.singletonList(BIDDER_CURRENCY))
                 .build();
     }
 
-    private static List<Imp> modifyImps(List<Imp> imps,
-                                        String masterTagId,
-                                        BigDecimal bidFloor,
-                                        String bidFloorCur) {
+    private static List<Imp> modifyImps(List<Imp> imps, String masterTagId, Price bidFloor) {
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:118]: floorPrice

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 6]
```diff
@@ -97,61 +87,45 @@ private ExtImpAdview parseExtImp(Imp imp) {
...
                     "Unable to convert provided bid floor currency from %s to %s for imp `%s`",
                     bidFloorCur, BIDDER_CURRENCY, impId));
         }
     }
 
-    private static BidRequest modifyRequest(BidRequest bidRequest,
-                                            String masterTagId,
-                                            BigDecimal bidFloor,
-                                            String bidFloorCur) {
+    private static BidRequest modifyRequest(BidRequest bidRequest, String masterTagId, Price bidFloor) {
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:111]: floorPrice

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 7]
```diff
@@ -97,61 +87,45 @@ private ExtImpAdview parseExtImp(Imp imp) {
...
+        return BidderUtil.isValidPrice(initialPrice)
+                ? convertBidFloor(initialPrice, imp.getId(), bidRequest)
+                : initialPrice;
     }
 
-    private BigDecimal convertBidFloor(BigDecimal bidFloor,
-                                       String bidFloorCur,
-                                       String impId,
-                                       BidRequest bidRequest) {
+    private Price convertBidFloor(Price bidFloor, String impId, BidRequest bidRequest) {
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:97]: floorPrice

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 8]
```diff
@@ -63,18 +63,8 @@ public AdviewBidder(String endpointUrl,
...
-                resolvedBidFloorCurrency = BIDDER_CURRENCY;
-                resolvedBidFloor = resolveBidFloor(request, firstImp, resolvedBidFloorCurrency);
-            } else {
-                resolvedBidFloorCurrency = firstImp.getBidfloorcur();
-                resolvedBidFloor = firstImp.getBidfloor();
-            }
-
-            modifiedBidRequest =
-                    modifyRequest(request, extImpAdview.getMasterTagId(), resolvedBidFloor, resolvedBidFloorCurrency);
+            final Price bidFloor = resolveBidFloor(firstImp, request);
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:66]: floorPrice

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java (hunk 4 above)
Comment (yevhenii-viktorov) [reply src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:91]: maybe even stick to `initialBidFloorPrice`

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java (hunk 5 above)
Comment (yevhenii-viktorov) [reply src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:118]: `bidFloorPrice`

**Review by snahornyi — APPROVED** [review]

**Pull request merged**